pkg/trace/obfuscate/sql.go

```diff
@@ -105,11 +107,39 @@ func (f *replaceFilter) Filter(token, lastToken TokenKind, buffer []byte) (token
 	switch token {
 	case String, Number, Null, Variable, PreparedStatement, BooleanLiteral, EscapeSequence:
 		return FilteredGroupable, []byte("?"), nil
+	case '?':
+		// Cases like 'ARRAY [ ?, ? ]' should be collapsed into 'ARRAY [ ? ]'
+		return FilteredGroupable, []byte("?"), nil
+	case Table:
+		if f.normalizeTables {
```

Comment:
Could we drop this field name and just access the config.HasFeature call here? Did you have something else in mind?